futures-test/src/assert.rs

```diff
@@ -2,16 +2,38 @@ use futures_core::stream::Stream;
 use std::marker::Unpin;
 
 #[doc(hidden)]
-pub fn assert_is_unpin_stream<S: Stream + Unpin>(_: &S) {}
+pub fn assert_is_unpin_stream<S: Stream + Unpin>(_: &mut S) {}
 
 /// Assert that the next poll to the provided stream will return
 /// [`Poll::Pending`][futures_core::task::Poll::Pending].
+///
+/// # Examples
+///
+/// ```
+/// #![feature(async_await, futures_api, pin)]
+///
+/// #[macro_use]
+/// extern crate futures_test;
+///
+/// #[macro_use]
+/// extern crate futures;
+///
+/// # fn main() {
```

Comment:
Main function doesn't need to be explicit

Reply:
It does when using `#[macro_use]`

Reply:
😏
https://github.com/rust-lang-nursery/futures-rs/blob/86c40ff7e0d5c28e37a14edd3732e25efdfb9805/futures-util/src/macros/pin.rs#L88-L94

Reply:
Ah, but only when `#[macro_use]` is on the same line as `extern crate`...

Reply:
Oh ^^' interesting